compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -337,7 +332,55 @@ trait QuotesAndSplices {
         new TreeTypeMap(typeMap = typeMap).transform(shape1)
       }
 
-    (typeBindings, shape2, patterns)
+    (bindSymMapping, shape2, patterns)
+  }
+
+  /** For each type variable defined in the quote pattern we generate an equivalent
+   *  binding that will be as type variable in the encoded `unapply` of the quote pattern.
+   *
+   *  @return Mapping from type variable symbols defined in the quote pattern into
+   *          type variable definitions for the `unapply` of the quote pattern.
+   *          This mapping retains the original type variable definition order.
+   */
+  private def unapplyBindingsMapping(quoted: Tree)(using Context): SeqMap[Symbol, Symbol] = {
+    // Collect all existing type variable bindings and create new symbols for them.
+    // The old info is used, it may contain references to the old symbols.
+    val (oldBindings, newBindings) = {
+      val seen = mutable.Set.empty[Symbol]
+      val oldBindingsBuffer = mutable.LinkedHashSet.empty[Symbol]
+      val newBindingsBuffer = mutable.ListBuffer.empty[Symbol]
```

Comment:
Now that we're returning a map from old bindings to new bindings without creating Bind nodes, I think the original idea of directly having a LinkedHashMap here makes more sense, sorry for the back and forth!